modules/dag-l1/src/test/scala/org/tessellation/dag/l1/infrastructure/trust/storage/TrustStorageSuite.scala

```diff
@@ -0,0 +1,52 @@
+package org.tessellation.dag.l1.infrastructure.trust.storage
+
+import cats.effect.IO
+
+import org.tessellation.dag.l1.config.types.DBConfig
+import org.tessellation.dag.l1.infrastructure.db.Database
+import org.tessellation.scalacheck.generators.trust.genTrustDbValues
+
+import ciris.Secret
+import eu.timepit.refined.auto._
+import weaver._
+
+object TrustStorageSuite extends SimpleIOSuite {
+
+  val testTrustDbValues = genTrustDbValues.sample.get
+
+  test("Test writing trust values to the database - score.") {
+    val dbConfig = DBConfig("org.sqlite.JDBC", "jdbc:sqlite::memory:", "sa", Secret(""))
+    Database.forAsync[IO](dbConfig).use { implicit db =>
```

Comment:
this creates a resource that can be provided at the suite level. please check the other tests with shared resources

Reply:
Changed the test to use a shared `Resource`, but not `ResourceSuite`.

I was unable to figure out how to write a test based on:

https://github.com/Constellation-Labs/tessellation/blob/001ab0b555165272704e3ec6c9e1febce32e2eb8/modules/dag-shared/src/test/scala/org/tessellation/dag/transaction/ContextualTransactionValidatorSuite.scala

so, I based my test on:

https://github.com/cngo-github/tessellation/blob/20220623-ngo-dev-trust-test/modules/dag-l1/src/test/scala/org/tessellation/dag/l1/domain/transaction/TransactionStorageSuite.scala

Reply:
I spent most of the day working and consulting with @ryle-goehausen about this. I’ve decided to keep using `SimpleIOSuite` because:

We cannot use a shared resource here because the database cannot be shared as the in-memory connection dropping leads to table dropping on all operations. Chaining any operations on top of the database resource leads to those operations not being re-run. There may be a way to get the chaining to work properly with writes, but all the things we tried didn’t work and it is not worth days of effort figuring it out.

The current implementation does reduce code repetition, and I think trying to jam the test into `ResourceSuite` will result in more complicated code.

Reply:
Got it. Let me try tomorrow